server/src/main/java/io/deephaven/server/session/SessionServiceGrpcImpl.java

```diff
@@ -345,7 +346,7 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(final ServerCall<Re
             if (altToken != null) {
                 try {
                     session = service.getSessionForToken(UUID.fromString(new String(altToken)));
-                } catch (IllegalArgumentException ignored) {
+                } catch (IllegalArgumentException | InvalidUuidException ignored) {
```

Comment:
We aren't using the f4b6a3 library for parsing here, so I don't think we should explicitly try to catch this exception; should we update to using f4b6a3 library? I guess wider Q: what's the difference between UUID and f4b6a3?

Reply:
The UuidCreator is capable of generating random uuids, unlike the default java impl.

According to its documentation, it is also substantially faster at parsing uuids than the default impl, so we're preferring it if we have the choice.

Reply:
In which case we should swap out the call from `UUID.fromString` to the `UuidCreator`?

Reply:
Changed to UuidCreator, and specified that the altToken bytes are ascii - but we need the IAE here as above in case the bytes aren't acsii? I'm not 100% that this is an invalid state though, just "we don't accept your token", so we move on?

Reply:
I don't think String throws IAE...

> This method always replaces malformed-input and unmappable-character sequences with this charset's default replacement string

Reply:
That is what the javadoc says, but doesn't quite match the bytecode I'm looking at (for Java 11, our default runtime for gradle). This might not strictly throw IAE for ascii, but there are paths to throw in this constructor:

`String(byte[])`  calls
`String(byte[], int, int)` calls
`java.lang.StringCoding#decode(byte[], int, int)`
* For utf8 this can end up in 
   `java.lang.StringCoding#decodeUTF8_0` which clearly calls `throwMalformed()`, which will be a IAE
* For ascii does seem to be safe
* for latin1 also seems to be safe
* for other charsets, this ends up in `java.lang.StringCoding.StringDecoder#decode`, which explicitly throws Error in response to `CharacterEncodingException`s

So I think you're right that this can't break, but I'm skeptical about trusting that it can't throw IAE at least in case we change our handling here (note that we used to decode as `file.encoding`, this patch fixes that to be ascii, and file.encoding defaults to utf-8 which according to the above _can_ throw IAE) or Java weakens its implementation.